README.md

```diff
@@ -147,68 +381,155 @@ For both these groups, containers present a unique challenge, as it seems almost
 
-Obviously, the greatest challenges to making the above a reality come down to security. Exposing any cluster resource to be accessible via continuous integration or version control triggers requires connecting the resource to external services, which always is a risk. Moving data from some external storage to a cluster resource to be analzyed is another challenge. A tiny step of progress is the ability to link a traditional HPC single sign on (SSO) scheme (e.g., LDAP or Kerberos) to a version control service (e.g., GitLab or GitHub) to allow for role based access and permissions but I've never tried or had access to this in practice (so I'm not sure if it's a good direction).
+Obviously, the greatest challenges to making the above a reality come down to security. Exposing any cluster resource to be accessible via continuous integration or version control triggers requires connecting the resource to external services, which always is a risk.
+
+> but this can be done effectively with modern auth -- a lot of internal HPC resources revolve
+> around kerberos and other antequated systms that do not really integrate with anything. IMO
+> modernizing this stuff to use things things token-based auth, limited life credentials, etc. is
+> really important and it's stuff we're thinking about. These secureity projects are opportunities
+> for collaboration in HPC too.
+
+> another gap for internal stuff that is not really covered here is storage services -- there are
+> way more options in the cloud, but HPC stroage is all shared filesystems. A *lot* of the security
+> issues come down to that. If you did not have shared filesystems, you could just have root and we
+> could isolate it. As it is, NFS3 and other legacy fileysstems used everywhere leave everyone's
+> data vulnerable if a single node is rooted. So looking at other types of storage -- or
+> selectively mounting filesystems via RMs more like EBS and other cloud storage is handled -- is
+> IMO a really good way to avoid thse types of issues.
+
+Moving data from some external storage to a cluster resource to be analzyed is another challenge. A tiny step of progress is the ability to link a traditional HPC single sign on (SSO) scheme (e.g., LDAP or Kerberos) to a version control service (e.g., GitLab or GitHub) to allow for role based access and permissions but I've never tried or had access to this in practice (so I'm not sure if it's a good direction).
+
+> what data are we talking about here? Not sure what this part is referring to. There is a lot of
```

Comment:
As an example, many domain sciences have large datasets that are publicly available (e.g., the Human Connectome Project) http://www.humanconnectomeproject.org/ and traditionally, step 1 to work with the data is to get it from somewhere (either sending physical disks or actually doing download from somewhere) to the HPC cluster. This is always a non-trivial task - and if there are two labs that are using the same data and they don't talk? It's done twice. And maybe it's thrown in a group space and not organized. At least speaking from the work I did 2009-2016, this was a constant problem.

Reply:
Most researchers don't come with an army of support and resources - they get a few thousand to buy computers and an account on the cluster and have to figure it out / make it work.